evaluator/package_cache.go

```diff
@@ -22,20 +26,43 @@ type PackageCacheMetadata struct {
 }
 
 type PackageCache struct {
-	mtx          sync.RWMutex
-	byID         map[int]*PackageCacheMetadata
-	byNevra      map[string]*PackageCacheMetadata
-	latestByName map[string]*PackageCacheMetadata
-	nameByID     map[int]string
+	enabled      bool
+	preload      bool
+	size         int
+	byID         *lru.Cache
+	byNevra      *lru.Cache
+	latestByName *lru.Cache
+	nameByID     *lru.Cache
 }
 
-func NewPackageCache() *PackageCache {
+func NewPackageCache(enabled bool, preload bool, size int) *PackageCache {
 	c := new(PackageCache)
-	c.byID = map[int]*PackageCacheMetadata{}
-	c.byNevra = map[string]*PackageCacheMetadata{}
-	c.latestByName = map[string]*PackageCacheMetadata{}
-	c.nameByID = map[int]string{}
-	return c
+
+	c.enabled = enabled
+	c.preload = preload
+	c.size = size
+
+	if c.enabled {
+		var err error
+		c.byID, err = lru.New(c.size)
+		if err != nil {
+			panic(err)
+		}
+		c.byNevra, err = lru.New(c.size)
+		if err != nil {
+			panic(err)
+		}
+		c.latestByName, err = lru.New(c.size)
```

Comment:
For latestByName and nameByID we need only package_name count size of cache, don't we?

Reply:
Yes, latestByName and nameByID will be smaller but in a corner case (every package have a different name) they can be as big as other caches. And because this param just states maximal size of cache I'd keep it same.

Reply:
Ok.